src/index.js

```diff
@@ -506,6 +528,55 @@ export default class AuthnWidget {
     }
   }
 
+  showDeviceManagementPopup(evt) {
+    // close other popups if any
+    var popupDivs = document.querySelectorAll("[id^='device-management-popup-frame']");
+    for (var i = 0; i < popupDivs.length; ++i) {
+      var popupDisplayStatus = popupDivs[i].style.display;
+      if (popupDisplayStatus === 'block') {
+        popupDivs[i].style.display = 'none';
+      }
+    }
+    let source = evt.currentTarget;
+    var deviceId = source.dataset['mfaSelectionKebabMenuContainer'];
+    var docId = 'device-management-popup-frame-' + deviceId;
+    document.querySelector("[id='" + CSS.escape(docId) + "']").style.display = 'block';
+  }
+
+  hideDeviceManagementPopup(evt) {
+    var target = evt.target;
+    var kebabMenuDivs = document.querySelectorAll("[id='kebab-menu-svg-id']");
+    for (var i = 0; i < kebabMenuDivs.length; ++i) {
+      if (kebabMenuDivs[i] == target) {
+        return;
+      }
+    }
+
+    var popupDivs = document.querySelectorAll("[id^='device-management-popup-frame']");
+    for (var i = 0; i < popupDivs.length; ++i) {
+      var popupDisplayStatus = popupDivs[i].style.display;
+      if (popupDisplayStatus === 'block') {
+        popupDivs[i].style.display = 'none';
+      }
+    }
+  }
+
+  handleMfaSetDefaultDeviceSelection(evt) {
+    evt.stopPropagation();
```

Comment:
Any reason to stop propagation rather than preventDefault? Sometimes its needed, but if not it can create troubleshooting problems later trying to figure out why an event doesn't fire

Reply:
This one has been tricky. I used stop propagation because of the device management popup out of kebab menu click, and expected UI behavior when user can do anything next from clicking on any other kebab menu, click somewhere else on the screen or on the device itself to trigger auth etc.  Reached my limits with JS and just used stop event propagation to quickly work it out. I hope I am not missing more things here.

Reply:
I'm okay with using it. I was thinking since we do a POST_FLOW below, the page will be re-rendered and whether we propagated the click or not shouldn't matter too much.

I'm okay with using stopPropagation if it's needed